test/e2e/localvolume_test.go

```diff
@@ -151,6 +159,269 @@ func LocalVolumeTest(ctx *framework.Context, cleanupFuncs *[]cleanupFn) func(*te
 		}
 		matcher.Expect(filepath.Base(pvs[0].Spec.Local.Path)).To(gomega.Equal(expectedPath))
 
+		// verify pv annotation
+		t.Logf("looking for %q annotation on pvs", provCommon.AnnProvisionedBy)
+		verifyProvisionerAnnotation(t, pvs, nodeList.Items)
+
+		// verify deletion
+		for _, pv := range pvs {
+			eventuallyDelete(t, &pv)
+		}
+		// verify that PVs come back after deletion
+		pvs = eventuallyFindPVs(t, f, localVolume.Spec.StorageClassDevices[0].StorageClassName, 1)
+
+		// consume pvs
+		consumingObjectList := make([]runtime.Object, 0)
+		for _, pv := range pvs {
+			pvc, job, pod := consumePV(t, ctx, pv)
+			consumingObjectList = append(consumingObjectList, job, pvc, pod)
+		}
+		// release pvs
+		eventuallyDelete(t, consumingObjectList...)
+
+		// verify that PVs eventually come back
+		matcher.Eventually(func() bool {
+
+			newPVs := eventuallyFindPVs(t, f, localVolume.Spec.StorageClassDevices[0].StorageClassName, 1)
+			for _, pv := range pvs {
+				pvFound := false
+				for _, newPV := range newPVs {
+					if pv.Name == newPV.Name {
+						if newPV.Status.Phase == corev1.VolumeAvailable {
+							pvFound = true
+						} else {
+							t.Logf("PV is in phase %q, waiting for it to be in phase %q", newPV.Status.Phase, corev1.VolumeAvailable)
+						}
+						break
+					}
+				}
+				// expect to find each pv
+				if !pvFound {
+					return false
+				}
+			}
+			return true
+		}, time.Minute*5, time.Second*5).Should(gomega.BeTrue(), "waiting for PVs to become available again")
+
+		// consume one PV
+		consumingObjectList = make([]runtime.Object, 0)
+
+		addToCleanupFuncs(cleanupFuncs, "pv-consumer", func(t *testing.T) error {
+			eventuallyDelete(t, consumingObjectList...)
+			return nil
+		})
+		for _, pv := range pvs[:1] {
+			pvc, job, pod := consumePV(t, ctx, pv)
+			consumingObjectList = append(consumingObjectList, job, pvc, pod)
+		}
+		// attempt localVolume deletion
+		matcher.Eventually(func() error {
+			t.Logf("deleting LocalVolume %q", localVolume.Name)
+			return f.Client.Delete(context.TODO(), localVolume, dynclient.PropagationPolicy(metav1.DeletePropagationBackground))
+		}, time.Minute*5, time.Second*5).ShouldNot(gomega.HaveOccurred(), "deleting LocalVolume %q", localVolume.Name)
+
+		// verify finalizer not removed with while bound pvs exists
+		matcher.Consistently(func() bool {
+			t.Logf("verifying finalizer still exists")
+			err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: localVolume.Name, Namespace: f.Namespace}, localVolume)
+			if err != nil && (errors.IsGone(err) || errors.IsGone(err)) {
```

Comment:
`errors.IsGone` is evaluated twice

Reply:
Fixed, it is now `(erros.IsGone(err) || errors.IsNotFound(err))`